pkg/plugin/resources.go

```diff
@@ -254,6 +275,133 @@ func (app *App) handleVectorSearch(w http.ResponseWriter, req *http.Request) {
 	w.Write(bodyJSON)
 }
 
+// SaveLLMOptInDataToGrafanaCom persists Grafana-managed LLM opt-in data to grafana.com.
+// This is required because provisioned plugins' settings do not survive a restart, and are
+// always reset to the state in grafana.com's provisioned-plugins endpoint.
+//
+// This function (and getPluginID) use the [/api/gnet][] endpoint, which [proxies requests][] to
+// the Grafana instance's configured [`GrafanaComUrl` setting][], which is set to the correct
+// value for the environment in Hosted Grafana instances.
+//
+// [/api/gnet]: https://github.com/grafana/grafana/blob/4d8287b319514b750617c20c130ffc424a3ecf2c/pkg/api/api.go#L677
+// [proxies requests]: https://github.com/grafana/grafana/blob/4bc582570ef7e713599ab3f2009fa75c27bb8a02/pkg/api/grafana_com_proxy.go#L28
+// [`GrafanaComUrl` setting]: https://github.com/grafana/grafana/blob/460be702619428e455ba74f8fb3bb563c1bea43a/pkg/setting/setting.go#L1088
+// Handles requests to /save-llm-state endpoint, and pushes state to GCom.
+func (app *App) handleSaveLLMState(w http.ResponseWriter, req *http.Request) {
+	log.DefaultLogger.Debug("Handling request to save LLM state to gcom..")
+	if app.saToken == "" {
+		// not available in Grafana < 10.2.3 or if externalServiceAccounts feature flag is not enabled
+		log.DefaultLogger.Warn("Service account token not available; cannot save LLM state to gcom")
+		return
+	}
+
+	if req.Method != http.MethodPost {
+		handleError(w, fmt.Errorf("invalid method"), http.StatusMethodNotAllowed)
+		return
+	}
+
+	// Read the request body
+	requestData := SaveLLMStateData{}
+	if req.Body != nil {
+		defer func() {
+			if err := req.Body.Close(); err != nil {
+				handleError(w, fmt.Errorf("failed to close request body %w", err), http.StatusInternalServerError)
+				return
+			}
+		}()
+		b, err := io.ReadAll(req.Body)
+		if err != nil {
+			handleError(w, fmt.Errorf("failed to read request body to bytes %w", err), http.StatusInternalServerError)
+			return
+		} else {
+			err := json.Unmarshal(b, &requestData)
+			if err != nil {
+				handleError(w, fmt.Errorf("failed to unmarshal request body to JSON %w", err), http.StatusInternalServerError)
+				return
+			}
+		}
+	} else {
+		log.DefaultLogger.Warn("Request body is nil")
+	}
+
+	notHG := os.Getenv("NOT_HG")
+	if notHG != "" {
+		log.DefaultLogger.Info("NOT_HG variable found; skipping saving settings to gcom")
+		return
+	}
```

Comment:
Could we do this check at the top of the function similar to the saToken check? I also think it needs error handling.

Reply:
Do we even need this logic? Isn't `NOT_HG` something we use for the ML plugin specifically

Reply:
Nvm, this logic is still needed for local dev. We need the option to not send to the gcom requests and just pass 200 so that the frontend save settings can pass without needing to set a valid gcom token